gradio-app/gradio
# Cache temp files created from base64 data

This PR slightly changes the way that we generate temp files from base64 input data by giving them a deterministic path based on the contents of data. In other words, if a user submits the same input twice, it will be produce the same temp file. If the temp file already exists, then the conversion from base64 to binary (which is very time consuming) can be skipped and the file can be returned directly. 

Fixes: #3189

Probably the easiest way to test this would be to run a demo like this:

```py
import gradio as gr

gr.Interface(lambda x:x, gr.Video(source="webcam"), gr.Video()).launch()
```

* Record a video that is at least a few seconds long
* Click the submit button (note the inference time)
* Click the submit button again with the same video (note the inference time, should be much faster on this branch)
